src/dmd/backend/cod2.c

```diff
@@ -2031,6 +2030,44 @@ void cdcond(CodeBuilder& cdb,elem *e,regm_t *pretregs)
         {
             // only sign-extension from 32-bits is available for 'and'
         }
+        else if (v1 == v2)
```

Comment:
this optimization is better done in cgelem.c

Reply:
To be honest I expected the front-end to eliminate such a trivial case, maybe it's worth to move such an optimization there (iif it can be proved the comparison is side-effect free).